CONTRIBUTING.md

```diff
@@ -0,0 +1,52 @@
+# Contributing
+
+## General guidelines and code of conduct
+
+This repository follows the same [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/digital-asset/daml/blob/master/CODE_OF_CONDUCT.md) as the [main DAML repository](https://github.com/digital-asset/daml/).
+
+## Build and test
+
+This project uses [NPM](https://npmjs.com/) to manage the dependencies and the environment of scripts handling the build lifecycle.
+
+Before you start coding, install the dependencies:
+
+    npm install
+
+Once this is done you can run the full build:
+
+    npm test
+
+The full build consists of:
+
+1. downloading the Ledger API ProtoBuffer service definitions from [Digital Asset's Bintray](https://bintray.com/digitalassetsdk/),
+2. generating the [gRPC](https://grpc.io/) stubs,
+3. running the [TypeScript](http://typescriptlang.org/) compilation, and finally
+4. running the tests using [Mocha](https://mochajs.org/).
+
+Steps 1 and 2 are executed only if the `proto` and `src/generated` directories are not available locally.
+
+To force these build phases to run again (e.g. because the bindings now depend on a new version of the gRPC Ledger API definitions), run the following before re-running `npm test`:
+
+    rm -rf proto src/generated
+
+## Release
+
+The following steps are to be followed in order to release a new version of the bindings:
+
+1.  Choose a new version number `<version>` (the `0.x` release series does not uphold semantic versioning)
+2.  Create the branch `release-<version>`
+3.  Change the title of the [unreleased section in the changelog](./CHANGELOG.md#Unreleased) to the chosen `<version>`
+4.  Commit the outstanding change (e.g. `git add CHANGELOG.md && git commit -m "Update changelog")
+5.  Run the command `npm version --no-git-tag-version <version>`
```

Comment:
This will create a second commit, and we want to squash afterwards anyway. I would suggest reordering the above steps to:
```
#!/usr/bin/env bash
set -euo pipefail
export VERSION=$1
## Maybe check we start from clean master here?
git checkout -b release-$VERSION
sed -i -E "/^## \\[Unreleased\\]$/a\\\n## v$VERSION\n" CHANGELOG.md
npm version --no-git-tag-version $VERSION
git add CHANGELOG.md package.json package-lock.json
git commit -m "release $VERSION"
```
Then `git push`.

Reply:
Thanks, I tried doing something like this yesterday but couldn't, I'll try to make this part of the release process at a later time.